swap_meet/vendor.py

```diff
@@ -1,2 +1,75 @@
+from operator import attrgetter
+
 class Vendor:
-    pass
\ No newline at end of file
+
+    def __init__(self, inventory=None):
+        self.inventory = inventory if inventory is not None else list() #using empty list constructor to explicitly note list data type expected
+
+    def add(self, item):
+        self.inventory.append(item) #appends instance of Item to inventory list
+        return item
+    
+    def remove(self, item):
+        if item not in self.inventory: #guard clause if item not in inventory
+            return False
+        
+        self.inventory.remove(item) #removes instance of Item from inventory list
+        return item
+    
+    def get_by_category(self, category):
+        return [item for item in self.inventory if item.category==category] #creates a list of all items in inventory whose category attribute matches category parameter
+    
+    def swap_items(self, friend, my_item, their_item):
+        if my_item not in self.inventory or their_item not in friend.inventory: #guard clause if either item not available to swap
+            return False
+
+        self.add(their_item)
+        friend.add(my_item)
+        self.remove(my_item)
+        friend.remove(their_item)
+        return True
+
+    def swap_first_item(self, friend):
+        if not self.inventory or not friend.inventory: #guard clause if either inventory is empty
+            return False
+
+        self.swap_items(friend, self.inventory[0], friend.inventory[0])
+        return True
+        
+    def get_best_by_category(self, category):
+        items_in_category = self.get_by_category(category)
+        if not items_in_category: #guard clause if category not available in inventory
+            return None
+        #accesses the values of the condition attribute in items_in_category and returns the item with the highest value
+        return max(items_in_category, key=attrgetter('condition'))
```

Comment:
Great use of `max` with the `get_by_category` function! If for some reason we didn't want to import the `operator` module for `attrgetter` we could also use `max` with a lambda function:
```suggestion
        return max(items_in_category, key=lambda item: item.condition)
```

Reply:
Interesting! Would be curious to chat through this in our next 1:1.

Reply:
Definitely, let's chat about it! Lambda functions can be really powerful in combination with functions that help us work with data like `max` and `filter`, so they're something you'll want to get comfortable with over time 🙂